### PR TITLE
Add module rainbow

### PR DESCRIPTION
**Pull request by tobes**
This is the most pointless yet most exciting module you can imagine.

Basically it allows color cycling of modules.  Imagine the joy of having the current time change through the colours of the rainbow.

If you were completely insane you could also use it to implement  the i3bar equivalent of the `<blink>` tag and cause yourself endless headaches and the desire to vomit.

### CODE REVIEW

**Comment by ultrabug** [comment]
lol @tobes you made me laugh out loud with this one :laughing: 

This project is becoming insane !! yet I'll review it asap since I love insanity ;)

**Comment by tobes** [comment]
:)

File: py3status/modules/rainbow.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+
+Configuration parameters:
+    cycle_time: How often we cdo a color change in seconds
+    (default 1)
+    force: If True then the color will always be set.  If false the color will
+        only be changed if it has not beeen set by a module.
+        (default False)
+    gradient: The colors we will cycle through, This is a list of hex values
+        (default [ '#FF0000', '#FFFF00', '#00FF00':q, '#00FFFF',
```
Comment (ultrabug) [inline py3status/modules/rainbow.py]: the `:q` looks like VIM command typo ?

**Comment by ultrabug** [comment]
I just tested it and I'm amazed by the result, it's **awesome** !!

I'll correct the minor doc typos and enhance the docstring a bit

**Pull request merged**